python/pynq/iop/grove_th02.py

```diff
@@ -0,0 +1,225 @@
+        ----------
+        if_id : int
+            IOP ID (1, 2, 3) corresponding to (PMODA, PMODB, ARDUINO).
+        gr_pin: list
+            A group of pins on stickit connector or arduino shield.
+            
+        """
+        if if_id in [PMODA, PMODB]:
+            if not gr_pin in [PMOD_GROVE_G3, \
+                              PMOD_GROVE_G4]:
+                raise ValueError("TH02 group number can only be G3 - G4.")
+            GROVE_TH02_PROGRAM = PMOD_GROVE_TH02_PROGRAM
+
+        elif if_id in [ARDUINO]:
+            if not gr_pin in [ARDUINO_GROVE_I2C]:
+                raise ValueError("TH02 group number can only be I2C.")
+            GROVE_TH02_PROGRAM = ARDUINO_GROVE_TH02_PROGRAM
+
+        else:
+            raise ValueError("No such IOP for grove device.")
+
+        self.iop = request_iop(if_id, GROVE_TH02_PROGRAM)
+        self.mmio = self.iop.mmio
+        self.log_interval_ms = 1000
+        self.log_running  = 0
+        self.iop.start()
+
+        if if_id in [PMODA, PMODB]:        
+            #: Write SCL and SDA Pin Config
+            self.mmio.write(iop_const.MAILBOX_OFFSET, gr_pin[0])
+            self.mmio.write(iop_const.MAILBOX_OFFSET+4, gr_pin[1])
+            
+        # Write configuration and wait for ACK
+        self.mmio.write(iop_const.MAILBOX_OFFSET + \
+                        iop_const.MAILBOX_PY2IOP_CMD_OFFSET, 1)
+        while (self.mmio.read(iop_const.MAILBOX_OFFSET + \
+                              iop_const.MAILBOX_PY2IOP_CMD_OFFSET) == 1):
+            pass  
+            
+    def read(self):
+        """Read the temperature and humidity values from the Grove TH02 
+        peripheral.
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        tuple
+            tuple containing (temperature, humidity)
+        
+        """
+        self.mmio.write(iop_const.MAILBOX_OFFSET+\
+                        iop_const.MAILBOX_PY2IOP_CMD_OFFSET, 2)      
+        while (self.mmio.read(iop_const.MAILBOX_OFFSET+\
+                                iop_const.MAILBOX_PY2IOP_CMD_OFFSET) == 2):
+            pass
+        tmp = self.mmio.read(iop_const.MAILBOX_OFFSET)
+        tmp = tmp/32 - 50
+
+        humidity = self.mmio.read(iop_const.MAILBOX_OFFSET + 0x4)
+        humidity = humidity/16 - 24
+        return (tmp, humidity)
+
+    def start_log(self, log_interval_ms = 100):
+        """Start recording multiple heart rate values in a log.
+        
+        This method will first call set the log interval before writting to
+        the MMIO.
+        
+        Parameters
+        ----------
+        log_interval_ms : int
+            The time between two samples in milliseconds.
+            
+        Returns
+        -------
+        None
+        
+        """
+        if (log_interval_ms < 0):
+            raise ValueError("Time between samples should cannot be less " + \
+                             "than zero.")
```

Comment:
Please fix the error message. (Also, try to use short message that can fit in 1 line).